src/utils/fixtures.js

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable import/prefer-default-export */
+/* eslint-disable no-param-reassign */
+import BigNumber from "bignumber.js";
+
+/**
+ * Generate a mock state with some slices and utxos with balances
+ * @param {Object} walletBalance
+ * @param {number} walletBalance.confirmedBalance
+ * @param {number} walletBalance.pendingBalance
+ */
+export function getMockState(walletBalance) {
+  const {
+    confirmedBalance,
+    pendingBalance,
+    numConfirmed = 5,
+    numPending = 2,
+  } = walletBalance;
+
+  const utxosConfirmed = Array(numConfirmed).fill({
+    confirmed: true,
+    amountSats: new BigNumber(Math.floor(confirmedBalance / numConfirmed), 10),
+    time: Date.now() - Math.floor(Math.random() * 10000),
+  });
+
+  const utxosUnconfirmed =
+    numPending && pendingBalance
+      ? Array(numPending).fill({
+          confirmed: false,
+          amountSats: new BigNumber(
+            Math.floor(pendingBalance / numPending),
+            10
+          ),
+        })
+      : [];
+
+  // we'll put half of the confirmed utxos in a deposit slice
+  // and half in change. All unconfirmed will go in one deposit slice
+  const state = {
+    wallet: {
+      deposits: {
+        nodes: [
+          {
+            utxos: utxosConfirmed.slice(Math.floor(utxosConfirmed.length / 2)),
+            change: false,
+          },
+          {
+            utxos: utxosUnconfirmed,
+            change: false,
+          },
+        ],
+      },
+      change: {
+        nodes: [
+          {
+            utxos: utxosConfirmed.slice(Math.ceil(utxosConfirmed.length / 2)),
+            change: true,
+          },
+        ],
+      },
+    },
+  };
+
+  // set slice balances
```

Comment:
I think we'll need to set more than just balances here:

```
[walden:~/code/caravan] add-tests(+34/-5)+ ± grep "isRequired" src/proptypes/slice.js
  address: PropTypes.string.isRequired,
  change: PropTypes.bool.isRequired,
  spend: PropTypes.bool.isRequired,
  fetchUTXOsError: PropTypes.string.isRequired,
  addressKnown: PropTypes.bool.isRequired,
```

I had to hack in the following in order to get the `ConfirmAddress` component to render in my test:

```js
  state.wallet.deposits.nodes.forEach((slice) => {
    slice.balanceSats = slice.utxos.reduce((balance, utxo) => {
      slice.address =
        "tb1qg73nnmlg3dcsfsfklkr3nu5xaw2u8d9hw8r0rchj8fs6m8s407uqu8qfa5";
      slice.spend = false;
      slice.fetchUTXOsError = "";
      slice.addressKnown = false;
      slice.multisig = {
        redeem: "",
        address:
          "tb1qg73nnmlg3dcsfsfklkr3nu5xaw2u8d9hw8r0rchj8fs6m8s407uqu8qfa5",
      };
      if (utxo.confirmed) return balance.plus(utxo.amountSats);
      return balance;
    }, new BigNumber(0));
  });
```
And other components will obviously require additional fields depending on functionality.

What do you think of grabbing the entire state object for each address type and use that as the fixture instead of creating it programmatically?

Reply:
actually, the address type is a good argument for continuing to do it programmatically. One state will not have multiple address types, so we'd probably want a store for each type, sometimes we might want to test the different types differently. address type and also `m-of-n` would also apply I'd think.

Reply:
Yeah, that's true, I guess we'd end up with a lot of copy/pasted fixtures to maintain.  I'm fine doing it the way you have it in the PR and hopefully we can spend some time untangling our state in the future.